### ISSUE TITLE
Stale MSBuild processes cause projects not to load

### ISSUE THREAD

**Issue by pkese** [ionide/ionide-vscode-fsharp#1256]
**Describe the bug**

I'm often getting situations where **Ionide fails to load solution**. This happens particularly when adding new projects to solution using `dotnet sln solution.sln add path/to/projects.fsproj`

The symptom is that projects won't load. All projects in solution explorer are marked with **(project not loaded yet)**

What I have noticed is that this is somehow related to having **multiple stale instances** of MSBuild process running even when vscode is closed:
```shell
$ ps -ef | grep dotnet
peterk   15199  3422  0 15:52 ?        00:00:03 /usr/share/dotnet/dotnet /usr/share/dotnet/sdk/3.0.100/MSBuild.dll /usr/share/dotnet/sdk/3.0.100/MSBuild.dll /nologo /nodemode:1 /nodeReuse:true
peterk   15240  3422  0 15:53 ?        00:00:02 /usr/share/dotnet/dotnet /usr/share/dotnet/sdk/3.0.100/MSBuild.dll /usr/share/dotnet/sdk/3.0.100/MSBuild.dll /nologo /nodemode:1 /nodeReuse:true
peterk   15267  3422  0 15:53 ?        00:00:02 /usr/share/dotnet/dotnet /usr/share/dotnet/sdk/3.0.100/MSBuild.dll /usr/share/dotnet/sdk/3.0.100/MSBuild.dll /nologo /nodemode:1 /nodeReuse:true
```

**After killing these dotnet processes** using `killall dotnet`, and restarting vscode, projects load normally.

I haven't noticed anything unusual in developer tools logs.  
`#F: Get Ionide logs` or `F#: Get info for diagnostics` don't work for me.

**To Reproduce**
It's hard to reproduce. Fiddle with solution file, load vscode, rinse&repeat until projects won't load, then close vscode and observe stale `dotnet MSBuild.dll` processes.

**Environment (please complete the following information):**
 - OS: Linux
 - Ionide version: 4.2.0
 - VSCode version: 1.39.2
 - dotnet SDK version: 3.0.100
 - mono / .Net Framework version: 6.4.0.198

**Comment by pkese** [comment]
I'm just guessing:  
It could be just that fiddling with .sln file somehow freezes those MSBuild.dll processes and causes them not to respond to Ionide requests any more.

**Comment by Krzysztof-Cieslak** [comment]
CC: @enricosada, any idea?

**Comment by superstator** [comment]
fwiw, I'm running into this pretty regularly. I will make a minor edit to some F# code, and on debugging it will instead run a previous build. Sometimes I can just restart VSCode and get back to work, but often I then hit the "not loaded yet" message. Once that happens I need to close all VSCode instances and run `killall -9 dotnet` before I can continue. I am not modifying the .sln when it happens, just code. I'd say I hit it once or twice a day, depending on how much F# hackery I'm engaging in.

Ubuntu 20.04 (5.4.0-48-generic)
VSCode 1.50.1
dotnet 3.1.402

Is there anything else I should look for that might help narrow this down?

**Comment by Krzysztof-Cieslak** [comment]
Should be fixed with 5.0
